File: src/main/java/cs/ut/controllers/training/AbstractModeController.java
package cs.ut.controllers.training;

import cs.ut.config.MasterConfiguration;
import cs.ut.config.items.ModelParameter;
import org.zkoss.zul.Vlayout;

import java.util.List;
import java.util.Map;

public abstract class AbstractModeController {
    protected Vlayout gridContainer;
    protected Map<String, List<ModelParameter>> parameters = MasterConfiguration.getInstance().getModelConfiguration().getProperties();

    protected AbstractModeController(Vlayout vlayout) {
        this.gridContainer = vlayout;
        parameters.remove(TrainingController.PREDICTION);
    }
}

File: src/main/kotlin/cs/ut/controllers/training/AbstractModeController.kt
package cs.ut.controllers.training

import cs.ut.config.MasterConfiguration
import cs.ut.config.items.ModelParameter
import cs.ut.controllers.TrainingController
import org.zkoss.zul.Vlayout

abstract class AbstractModeController(protected val gridContrainer: Vlayout) {
    protected val parameters: Map<String, List<ModelParameter>> by lazy {
        (MasterConfiguration.modelConfiguration.properties - TrainingController.PREDICTION)
    }
}